tests/core/models/test_edition.py

```diff
@@ -325,6 +325,20 @@ def test_calculate_presentation_author(self, db: DatabaseTransactionFixture):
         assert "Kelly Accumulator, Bob A. Bitshifter" == wr.author
         assert "Accumulator, Kelly ; Bitshifter, Bob" == wr.sort_author
 
+    def test_calculate_presentation_very_long_author(
+        self, db: DatabaseTransactionFixture
+    ):
+        authors = []
+        for i in range(0, 500):
+            author, ignore = db.contributor(sort_name=f"AuthorLast{i}, AuthorFirst{i}")
+            authors.append(author.sort_name)
+        wr = db.edition(authors=authors)
+        author, sort_author = wr.calculate_author()
+        wr.calculate_presentation()
```

Comment:
To ensure that this test is valid, I think we need an assertion on the lengths before they are truncated, in addition to the ones after. Probably should check a couple of different scenarios, as well:
- length > max length: assert truncated to max length and ends in "..."
- length = max length (and doesn't end in "..."): assert truncated to max length and doesn't end in ...
- length < max length: assert truncated == original

Reply:
I will fix.

Reply:
Do we have a test to ensure that the ellipses does not get added to strings with length less than or equal to the `Edition.SAFE_AUTHOR_FIELD_LENGTH_TO_AVOID_PG_INDEX_ERROR`?

Reply:
will add.